xbmc/guilib/GUIIncludes.cpp

```diff
@@ -513,13 +513,22 @@ std::string CGUIIncludes::ResolveConstant(const std::string &constant) const
   return StringUtils::Join(values, ",");
 }
 
-std::string CGUIIncludes::ResolveExpressions(const std::string &expression) const
+std::string CGUIIncludes::ResolveExpressions(const std::string &strInput, const std::vector<std::string> &resolved /* = std::vector<std::string>() */) const
```

Comment:
Thinking on it further, it is probably a good idea to flatten nested expressions after they are loaded up here [L218](https://github.com/rmrector/xbmc/blob/ce501e0d1eb4aaaf979cef455269a3e736060a42/xbmc/guilib/GUIIncludes.cpp#L218), rather than recursing through expressions for every conditional statement in the skin.